src/mmrelay/setup_utils.py

```
@@ -359,41 +483,48 @@ def service_needs_update():
-        bool: True if user lingering is enabled, False otherwise.
+    Checks for a usable `loginctl` executable, queries `loginctl show-user <user> --property=Linger`
+    (using the environment variable USER or USERNAME to determine the account), and returns True
+    only if the command succeeds and reports `Linger=yes`. If `loginctl` is not found, the command
+    fails, or an unexpected error occurs, the function returns False.
     """
     try:
         username = os.environ.get("USER", os.environ.get("USERNAME"))
+        loginctl = shutil.which("loginctl")
+        if not loginctl:
+            return False
         result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)